pkg/storage/pebble.go

```diff
@@ -1908,8 +1913,7 @@ func pebbleExportToSst(
 					return roachpb.BulkOpSummary{}, MVCCKey{}, errors.Wrapf(err, "adding key %s", unsafeKey)
 				}
 			}
-			newSize := curSize + int64(len(unsafeKey.Key)+len(unsafeValue))
-			if maxSize > 0 && newSize > int64(maxSize) {
+			if reachedMaxSize {
```

Comment:
I think we can pull this up to 1905 -- as soon as we know we're over side and _not_ paginating, we can bail without proceeding to add to the sst writer first